CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+# Project architecture
+
+The project is divided (as of now) into two main sections: *backend* and *frontend*, each with their quirks.
+
+## Backend architecture
+
+We use [Clojure](https://www.braveclojure.com) as the backend language for pretty much **everything** due to basically two reasons:
+
+1. Because of the "everything is immutable" paradigm, it's very easy to reason about the logic of the code, which will probably get very complex as time goes on.
+2. Due to its functional nature, it is *very* easy to run tests on it and a strict TDD discipline.
+
+The backend also uses [Redis](https://redis.io) as a key-value storage mechanism for any persistent information necessary (such as the state of a game) since we will *always* know exactly what key we're searching for (we'll know the game ID, player ID, etc.).
+
+Finally, the backend uses [Compojure](https://github.com/weavejester/compojure) to define it's [REST API](http://www.restapitutorial.com), which is the way the *frontend* will communicate with it.
+
+## Frontend architecture
+
+Frontend uses a standard [Vue.js](https://vuejs.org) architecture due to being easy to set up and not overcomplicating the frontend aspects.
+
+# Project structure
+
+The structure of a project (where the f*** is that file?) might be more important than it's architecture. In this case, there is a three-way split: *backend* and *frontend* as before, but we also have de *e2e* tests to worry about.
+
+But some things also happen at the root:
+
+- **docker-compose.yml**: Defines how the different parts of the project launch and interact with each other.
+- **.travis.yml**: Defines how the different pars of the project are to be tested.
+
+However, we will understand this further in the following sections.
+
+## Backend structure
+
+- **backend/Dockerfile**: Defines how the Docker image for the backend will be built and run
+- **backend/project.clj**: Defines the project dependencies and entry point
+- **backend/test/card_game/**: Contains all the unit tests for the code. Remember that a unit test must be done *before* the actual code.
+- **backend/src/card_game/**: Contains the code that will run in production, the actual logic.
+- **backend/src/card_game/persistence.clj**: This is the only file to interact directly with the database. All database calls *must* pass through here.
+- **backend/src/card_game/api_handler.clj**: This is the only file to interact directly with frontend. Any actions to take as a response to a frontend call *must* be defined here.
+
+## Frontend structure
+
+- **frontend/Dockerfile** and **frontend/entry.sh**: Defines the Docker image and its entrypoint.
```

Comment:
Same as before: What's a *Docker image* and an *entrypoint* (maybe I should know, sorry for that)